frontend/src/metabase/components/ExplicitSize/ExplicitSize.tsx

```diff
@@ -47,12 +47,15 @@ type InnerProps = {
 
 type ExplicitSizeOuterProps<T> = Omit<T, "width" | "height">;
 
+/**
+ * @deprecated HOCs are deprecated
+ */
 function ExplicitSize<T>({
   selector,
   wrapped = false,
   refreshMode = "throttle",
 }: ExplicitSizeProps<T> = {}) {
-  return (ComposedComponent: ComponentType<T & SizeState>) => {
```

Comment:
State is not Props

Reply:
That is not what this type is claiming. It claims that the function wrapped by `ExplicitSize({...})` will receive it's preexisting props, _plus_ `SizeState` (which is true as per [line 238](https://github.com/metabase/metabase/pull/49949/files#diff-85a9ce289a36d34260151add8324b1d106eb90fc50c2079f749811af0b51485aL238).

That means we can write:

```ts
type MyProps = {
  foo: string
}

ExplicitSize<MyProps>({ ... })(function (props) {
  console.log(props.foo)

  // these exist and TS knows about them, even though we did not specify them in MyProps 
  console.log(props.width)
  console.log(props.height)
}
```

Reply:
Good catch!
Fixed: 5d29a45cd73b3be7cfc6954c224f231df4734f42